File: src/leetcode/invertbinarytree/InvertBinaryTree.java
package leetcode.invertbinarytree;

import trees.TreeNode;

public class InvertBinaryTree {

    TreeNode invert(TreeNode root) {

        doInvert(root);
        return root;
    }

    void doInvert(TreeNode root) {

        if (root != null) {
            TreeNode tmp = root.left;
            root.left = root.right;
            root.right = tmp;

            doInvert(root.left);
            doInvert(root.right);
        }
    }
}

File: python-code/neetcode150/invert_binary_tree.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None):
        self.val = val
        self.left = left
        self.right = right



def invert_tree(root:TreeNode | None)->TreeNode | None:
    if root is None:
        return root
    
    # Invert subtrees
    root.left = invert_tree(root.left)
    root.right = invert_tree(root.right)

    # Invert current level
    tmp = root.left
    root.left = root.right
    root.right = tmp
    return root